src/jsifier.js

```diff
@@ -244,7 +244,9 @@ function JSify(data, functionsOnly) {
       } else if (typeof snippet === 'function') {
         isFunction = true;
         snippet = processLibraryFunction(snippet, ident, finalName);
-        Functions.libraryFunctions[finalName] = 1;
+        if (ident[0] != '$') { // Functions that start with '$' should not be imported to asm.js/wasm module
+          Functions.libraryFunctions[finalName] = 1;
+        }
```

Comment:
Are Functions.libraryFunctions only used for exporting to the native code?  In which case yes this looks good.  I guess we should have always being doing this, but because native symbosl can't start with `$` no C/C++ code could ever bind to them?

Would you mind making this a separate PR?    I think its worth landing on its own and I'm curious if it effects any test output?

Reply:
If possible I'd prefer to keep git branch wrangling to a minimum - this change is after all needed to implement the behavior needed by this PR?

Reply:
OK.  I'm mostly curious if this change effect any tests.  Do you know if it does?

Reply:
Ah yes I see that it does.  It looks like it effect the list of "sent" functions that we report for all fastcomp tests.  I believe this is because with fastcomp we were needless exporting all these functions to native code.   These functions are still included in the build, but they are no longer part of the exported function list.. which is great.  I think its is a genuine code size win for fastcomp builds !  Nice!    (which is kind of why I think it should be a separate change ideally).

Reply:
Why do we check the name - can't  we use the `__import` note?

Reply:
Because we implicitly import all functions.. we don't want to have to label all our functions with `__import` do we?    It looks like `__import` is for things that we want to pretent are functions but are actually just strings.

Reply:
maybe `__import` should have been called `__fake_function` or `__is_actually_function` :)

Reply:
I see, thanks @sbc100